pkg/ebpf/events_pipeline.go

```diff
@@ -252,6 +259,24 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 			evt.Syscall = syscall
 			evt.Metadata = nil
 
+			// Create unique identifier for both, the task and its parent. This will allow the
+			// engine to correlate events from the same task and from the same parent task. The
+			// identifier will never be the same even if the task is reused (e.g. a process forked
+			// and the child process reused the same PID).
+			evt.ProcIdentifier = trace.ProcUniqueIdentifier{
```

Comment:
Regular events will be able to update the process tree using the "process hash" calculated here (fast hashing, using pid + starttime).